src/custom/hooks/web3.ts

```diff
@@ -125,6 +126,25 @@ export function useEagerConnect() {
     return () => timeout && clearTimeout(timeout)
   }, [active])
 
+  useEffect(() => {
+    // fix for this https://github.com/gnosis/cowswap/issues/1923
+    // check if current connector is of type WalletConnect
+    if (connector instanceof WalletConnectConnector) {
+      const walletConnect = connector.walletConnectProvider.signer.connection.wc
+
+      // listen on disconnect events directly on WalletConnect client and close connection
+      // important if the connection is closed from the wallet side after page refresh
+      walletConnect.on('disconnect', (error: any) => {
+        connector.close()
+        localStorage.removeItem(STORAGE_KEY_LAST_PROVIDER)
+
+        if (error) {
+          throw error
```

Comment:
What happens if the `on` function receives an error? u are bubling up, is this intended?

Reply:
Not sure where I was going with this one but I've changed it to just console log if there is an error otherwise execute rest of the that block code.